File: algorithms/easy/delete_greatest_value_in_each_row.cpp
class Solution {
public:
    int deleteGreatestValue(vector<vector<int>>& grid) {
      for (int i = 0; i < grid.size(); ++i) {
        sort(grid[i].begin(), grid[i].end());
      }
      int ans = 0;
      for (int j = 0; j < grid[0].size(); ++j) {
        int tmp = 0;
        for (int i = 0; i < grid.size(); ++i) {
          tmp = max(tmp, grid[i][j]);
        }
        ans += tmp;
      }
      return ans;
    }
};

File: algorithms/easy/delete_greatest_value_in_each_row.ex
defmodule Solution do
  @spec delete_greatest_value(grid :: [[integer]]) :: integer
  def delete_greatest_value(grid) do
    grid
    |> Enum.map(fn row -> Enum.sort(row, &>=/2) end)
    |> Enum.zip()
    |> Enum.map(fn tp ->
      tp
      |> Tuple.to_list()
      |> Enum.max()
    end)
    |> Enum.sum()
  end
end
